File: 6/ts/Jet.ts
/*
 * The Jet Class
 *
 * author Dominic M.
 * version 1.0
 * since 2024-10-16
 */

import { Airplane } from './Airplane.ts'

export class Jet extends Airplane {
  /**
   * The speed of plane
   */
  private readonly MULTIPLYER: number = 2

  /**
   * Sets the speed by multipying inputted speed by multiplyer
   *
   * @param speed of input
   */
  public setSpeed (speed: number): void {
    super.theSpeed = speed * this.MULTIPLYER
  }

  /**
   * Sets the speed by multiplying the speed by 2
   */
  public accelerate (): void {
    this.theSpeed = this.theSpeed * 2
  }
}

File: 6/java/Jet.java
/**
 * The Jet Class.
 *
 * @author Dominic M.
 * @since 2024-9-22
 * @version 1.0
 */

/**
 * This is the Jet Class
 */
public class Jet extends Airplane {
    /**
     * The speed of the plane.
     */
    private static final int MULTIPLYER = 2;

    /**
     * This is the constructor.
     */
    public Jet() {
        super();
    }

    /**
     * Sets the speed by mutiplying the inputted speed by the multiplyer.
     */
    public void setSpeed(int speed) {
        super.setSpeed(speed * MULTIPLYER);
    }

    /**
     * Sets the speed by doubling the speed.
     */
    public void accelerate() {
        super.setSpeed(getSpeed() * 2);
    }
}
